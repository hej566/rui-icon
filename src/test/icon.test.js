import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Icon from '../component';
import Acorn1, { ReactComponent as Acorn2 } from '../assets/svg/acorn-duotone.svg';

const propsForImgSvg = {
  height: '2rem',
  width: '2rem',
  color: 'blue',
  src: Acorn1,
  alt: 'svg image',
  motion: 'rotate 0.5s',
  className: 'test class',
};

const propsForSvgObj = {
  height: '2rem',
  width: '2rem',
  color: 'blue',
  motion: 'rotate 0.5s',
  className: 'test class',
};

it('IMG SVG render correctly', () => {
  const tree = renderer.create(<Icon src={Acorn1} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('OBJ SVG render correctly', () => {
  const tree = renderer.create(<Icon>{Acorn2}</Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('IMG SVG render correctly with height, width, color, motion, classname', () => {
  const IconComponent = <Icon {...propsForImgSvg} />;
  const tree = renderer.create(IconComponent).toJSON();
  expect(tree).toMatchSnapshot();
});

it('OBJ SVG render correctly with height, width, color, motion, classname', () => {
  const IconComponent = <Icon {...propsForSvgObj}>{Acorn2}</Icon>;
  const tree = renderer.create(IconComponent).toJSON();
  expect(tree).toMatchSnapshot();
});

it('IMG props with height, width, color and motion', () => {
  const IconComponent = <Icon {...propsForImgSvg} />;
  const testRender = renderer.create(IconComponent);
  const testInstance = testRender.root;
  expect(testInstance.props).toEqual({ ...propsForImgSvg });
});

it('IMG click event', () => {
  const clickHandler = jest.fn();
  const IconComponent = <Icon {...propsForImgSvg} onClick={clickHandler} />;
  render(IconComponent);
  const element = screen.getByRole('figure');
  fireEvent.click(element);
  expect(clickHandler.mock.calls.length).toEqual(1);
});

it('IMG props with height, width, color and motion', () => {
  const IconComponent = <Icon {...propsForSvgObj}>{Acorn2}</Icon>;
  const testRender = renderer.create(IconComponent);
  const testInstance = testRender.root;
  expect(testInstance.props).toEqual({ ...propsForSvgObj, children: Acorn2 });
});

test('IMG click event', () => {
  const clickHandler = jest.fn();
  const IconComponent = (
    <Icon {...propsForSvgObj} onClick={clickHandler}>
      {Acorn2}
    </Icon>
  );
  render(IconComponent);
  const element = screen.getByRole('figure');
  fireEvent.click(element);
  expect(clickHandler.mock.calls.length).toEqual(1);
});
