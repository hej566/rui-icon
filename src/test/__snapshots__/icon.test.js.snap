// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IMG SVG render correctly 1`] = `
<div
  className="sc-bdvvtL jFcxMn"
  onClick={[Function]}
  role="figure"
>
  <i
    className="sc-gsDKAQ kBvhNT"
  >
    <img
      alt=""
      className="sc-dkPtRN gvNhlX"
      src="SvgrURL"
    />
  </i>
</div>
`;

exports[`IMG SVG render correctly with height, width, color, motion, classname 1`] = `
<div
  className="sc-bdvvtL jFcxMn test class"
  onClick={[Function]}
  role="figure"
>
  <i
    className="sc-gsDKAQ dZSVlC"
  >
    <img
      alt="svg image"
      className="sc-dkPtRN gvNhlX"
      src="SvgrURL"
    />
  </i>
</div>
`;

exports[`OBJ SVG render correctly 1`] = `
<div
  className="sc-bdvvtL jFcxMn"
  onClick={[Function]}
  role="figure"
>
  <i
    className="sc-gsDKAQ kBvhNT"
  >
    
    <svg
      color="currentColor"
      height="1rem"
      role="none"
      width="1rem"
    />
  </i>
</div>
`;

exports[`OBJ SVG render correctly with height, width, color, motion, classname 1`] = `
<div
  className="sc-bdvvtL jFcxMn test class"
  onClick={[Function]}
  role="figure"
>
  <i
    className="sc-gsDKAQ dZSVlC"
  >
    
    <svg
      color="blue"
      height="2rem"
      role="none"
      width="2rem"
    />
  </i>
</div>
`;
